vxquery-core/src/main/java/org/apache/vxquery/compiler/rewriter/rules/PushChildIntoDataScanRule.java

```diff
@@ -120,10 +118,15 @@ private boolean updateDataSource(VXQueryCollectionDataSource ds, Mutable<ILogica
         ExpressionToolbox.findAllFunctionExpressions(expression, BuiltinOperators.CHILD.getFunctionIdentifier(), finds);
         for (int i = finds.size(); i > 0; --i) {
             int typeId = ExpressionToolbox.getTypeExpressionTypeArgument(finds.get(i - 1));
+            SequenceType sType = null;
```

Comment:
I don't think you need to declare the variable here. Either move it to be a class variable or move it inside the if statement.